Sources/DiffableDataSource/Logger.swift

```diff
@@ -0,0 +1,18 @@
+//
+//  Logger.swift
+//  
+//
+//  Created by Jaime Azevedo on 02/04/2021.
+//
+
+import Foundation
+
+final class Logger {
+    private(set) static var shared = Logger()
+
+    var logBlock: ((DiffableError) -> Void)?
+
+    func log(error: DiffableError) {
+        logBlock?(error)
+    }
+}
```

Comment:
I was thinking something a bit simpler for the logging with a global function and for configuring using a more general method for setting up the Dependencies. 
```suggestion
public struct Dependency {
    public static var logFault: ((DiffableError) -> Void) = { _ in }
}

func logFault(error: DiffableError) {
    Dependency.logFault(error)
}
```
which could be configured like this:
```
DiffableDataSource.Dependency.logFault = logFault
```

Reply:
Sounds good, I've committed this changes with file renaming.